lambda-streaming-ttfb-write-sam-with-dynamodb/README.md

```diff
@@ -0,0 +1,90 @@
+- [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html) installed and configured
+- [Git Installed](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+- [AWS Serverless Application Model](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html) (AWS SAM) installed
+
+## Deployment Instructions
+
+1. Create a new directory, navigate to that directory in a terminal and clone the GitHub repository:
+
+```
+git clone https://github.com/aws-samples/serverless-patterns
+```
+
+1. Change directory to the pattern directory:
+
+   ```
+   cd lambda-streaming-ttfb-write-sam-with-dynamodb
+   ```
+
+1. From the command line, use AWS SAM to deploy the AWS resources for the pattern as specified in the template.yml file:
+
+   ```
+   sam deploy -g --stack-name lambda-streaming-ttfb-write-sam-with-dynamodb
+   ```
+
+1. During the prompts:
+
+   - Enter a stack name
+   - Enter the desired AWS Region - AWS CLI default region is recommended if you are planning to run the test (test.sh) script
+   - Allow SAM CLI to create IAM roles with the required permissions.
+
+1. Once you have run `sam deploy --guided` mode once and saved arguments to a configuration file `samconfig.toml`, you can use `sam deploy` in future to use these defaults.
+
+AWS SAM deploys a Lambda function with streaming support and a function URL
+
+![AWS SAM deploy --g](https://d2908q01vomqb2.cloudfront.net/1b6453892473a467d07372d45eb05abc2031647a/2023/03/31/AWS-SAM-deploy-g.png)
+
+The AWS SAM output returns a Lambda function URL.
+
+![AWS SAM resources](https://d2908q01vomqb2.cloudfront.net/1b6453892473a467d07372d45eb05abc2031647a/2023/03/31/AWS-SAM-resources.png)
+
+## How it works
+
+The service interaction in this pattern uses AWS Lambda's response streaming capability to stream data in batches from Amazon DynamoDB. Instead of retrieving all query / scan results at once, the Lambda function processes the query incrementally, retrieving and sending results as they become available. Here are the details of the interaction:
+
+1. Client Request
+   A client sends a request to the Lambda URL, asking for specific data from a DynamoDB table.
+2. Lambda Function Initialization :
+   When the Lambda function is invoked, it initializes a connection to DynamoDB and prepares to run a query operation. The query includes a limit parameter, which restricts the number of items retrieved in each batch.
+3. Querying
+   The Lambda function begins querying DynamoDB using the defined limit (e.g., 100 items per batch). DynamoDB will return only a limited set of results instead of the entire result set at once.
+4. Response Streaming
+   As soon as a batch of results is retrieved, the Lambda function uses the `write()` method of Lambda's streaming API to send the data to the client. This happens immediately, without waiting for the entire query operation to complete.
```

Comment:
```suggestion
   As soon as a batch of results is retrieved, the function uses Lambda's streaming API `write()` method to send the data to the client. This happens immediately, without waiting for the entire query operation to complete.
```

Reply:
Applied the suggested changes in the latest commit.